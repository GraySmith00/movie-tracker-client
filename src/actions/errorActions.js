export const setRegisterErrorState = errorMessage => ({
  type: 'SET_REGISTER_ERROR_STATE',
  errorMessage
});

export const setLoginErrorState = errorMessage => ({
  type: 'SET_LOGIN_ERROR_STATE',
  errorMessage
});

export const setFavoritesErrorState = errorMessage => ({
  type: 'SET_FAVORITES_ERROR_STATE',
  errorMessage
});
