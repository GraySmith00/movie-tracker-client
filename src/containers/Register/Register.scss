.register-form {
  display: flex;
  flex-direction: column;
  width: 100%;
  justify-content: center;
  align-items: center;
  margin: 2.5rem 0 0 0;

  h3 {
    font-size: 3rem;
    margin: 0 0 1rem 0;
  }
}

.register-name {
  height: 3rem;
  width: 25rem;
  font-size: 1rem;
  margin-bottom: 1.5rem;
  border-radius: 0.5rem;
  outline: none;
  border: 1px #00aad5 solid;
  padding-left: 1rem;
}

.register-email {
  height: 3rem;
  width: 25rem;
  font-size: 1rem;
  margin-bottom: 1.5rem;
  border-radius: 0.5rem;
  outline: none;
  border: 1px #00aad5 solid;
  padding-left: 1rem;
}

.register-password {
  height: 3rem;
  width: 25rem;
  font-size: 1rem;
  margin-bottom: 1.5rem;
  border-radius: 0.5rem;
  outline: none;
  border: 1px #00aad5 solid;
  padding-left: 1rem;
}

.register-btn {
  background-color: #00aad5;
  height: 3rem;
  width: 25rem;
  font-size: 1.5rem;
  border-radius: 0.5rem;
  color: #fff;
}

.error-message {
  margin: 0 0 0.5rem 0;
  text-align: center;
  color: crimson;
}
