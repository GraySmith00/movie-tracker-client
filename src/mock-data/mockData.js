const nowPlayingResultsArray = {
  results: [
    {
      vote_count: 507,
      id: 345940,
      video: false,
      vote_average: 6.2,
      title: "The Meg",
      popularity: 173.315,
      poster_path: "/xqECHNvzbDL5I3iiOVUkVPJMSbc.jpg",
      original_language: "en",
      original_title: "The Meg",
      genre_ids: [28, 878, 53, 27],
      backdrop_path: "/ibKeXahq4JD63z6uWQphqoJLvNw.jpg",
      adult: false,
      overview:
        "A deep sea submersible pilot revisits his past fears in the Mariana Trench, and accidentally unleashes the seventy foot ancestor of the Great White Shark believed to be extinct.",
      release_date: "2018-08-09"
    },
    {
      vote_count: 1095,
      id: 353081,
      video: false,
      vote_average: 7.4,
      title: "Mission: Impossible - Fallout",
      popularity: 138.478,
      poster_path: "/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg",
      original_language: "en",
      original_title: "Mission: Impossible - Fallout",
      genre_ids: [12, 28, 53],
      backdrop_path: "/5qxePyMYDisLe8rJiBYX8HKEyv2.jpg",
      adult: false,
      overview:
        "When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfil his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.",
      release_date: "2018-07-25"
    },
    {
      vote_count: 2833,
      id: 351286,
      video: false,
      vote_average: 6.6,
      title: "Jurassic World: Fallen Kingdom",
      popularity: 102.979,
      poster_path: "/c9XxwwhPHdaImA2f1WEfEsbhaFB.jpg",
      original_language: "en",
      original_title: "Jurassic World: Fallen Kingdom",
      genre_ids: [28, 12, 878],
      backdrop_path: "/3s9O5af2xWKWR5JzP2iJZpZeQQg.jpg",
      adult: false,
      overview:
        "Several years after the demise of Jurassic World, a volcanic eruption threatens the remaining dinosaurs on the island of Isla Nublar. Claire Dearing, the former park manager and founder of the Dinosaur Protection Group, recruits Owen Grady to help prevent the extinction of the dinosaurs once again.",
      release_date: "2018-06-06"
    }
  ]
};

export const modifiedNowPlaying = [
  {
    id: 345940,
    title: "The Meg",
    releaseDates: "2018-08-09",
    overview:
      "A deep sea submersible pilot revisits his past fears in the Mariana Trench, and accidentally unleashes the seventy foot ancestor of the Great White Shark believed to be extinct.",
    poster_path: "/xqECHNvzbDL5I3iiOVUkVPJMSbc.jpg"
  },
  {
    id: 353081,
    title: "Mission: Impossible - Fallout",
    img: "/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg",
    overview:
      "When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfil his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.",
    releaseDates: "2018-07-25"
  },
  {
    id: 351286,
    title: "Jurassic World: Fallen Kingdom",
    img: "http://image.tmdb.org/t/p/original/c9XxwwhPHdaImA2f1WEfEsbhaFB.jpg",
    realeaseDates: "2018-06-06",
    overview:
      "Several years after the demise of Jurassic World, â€¦event the extinction of the dinosaurs on the island of Isla Nublar. Claire Dearing, the former park manager and founder of the Dinosaur Protection Group, recruits Owen Grady to help prevent the extinction of the dinosaurs once again."
  }
];
