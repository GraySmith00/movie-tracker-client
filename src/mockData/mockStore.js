export const mockStore = {
  movies: {
    nowPlaying: [
      {
        movie_id: 345940,
        title: 'The Meg',
        release_date: '2018-08-09',
        overview:
          'A deep sea submersible pilot revisits his past fears in the Mariana Trench, and accidentally unleashes the seventy foot ancestor of the Great White Shark believed to be extinct.',
        vote_average: 6.2,
        poster_path:
          'http://image.tmdb.org/t/p/original/xqECHNvzbDL5I3iiOVUkVPJMSbc.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 353081,
        title: 'Mission: Impossible - Fallout',
        release_date: '2018-07-25',
        overview:
          'When an IMF mission ends badly, the world is faced with dire consequences. As Ethan Hunt takes it upon himself to fulfil his original briefing, the CIA begin to question his loyalty and his motives. The IMF team find themselves in a race against time, hunted by assassins while trying to prevent a global catastrophe.',
        vote_average: 7.3,
        poster_path:
          'http://image.tmdb.org/t/p/original/AkJQpZp9WoNdj7pLYSj1L0RcMMN.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 402900,
        title: "Ocean's Eight",
        release_date: '2018-06-07',
        overview:
          "Debbie Ocean, a criminal mastermind, gathers a crew of female thieves to pull off the heist of the century at New York's annual Met Gala.",
        vote_average: 7,
        poster_path:
          'http://image.tmdb.org/t/p/original/MvYpKlpFukTivnlBhizGbkAe3v.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 507569,
        title: 'The Seven Deadly Sins: Prisoners of the Sky',
        release_date: '2018-08-18',
        overview:
          'The Seven Deadly Sins travel to a remote land in search of the phantom ingredient "sky fish." Meliodas and Hawk end up at a "Sky Palace" that exists above the clouds, where all the residents have wings. Meliodas is mistaken for a boy who committed a crime and is thrown in prison. Meanwhile, the residents are preparing a ceremony for defense against a ferocious beast that awakens once every 3,000 years. But the Six Knights of Black, a Demon Clan army, arrives and removes the seal on the beast in order to threaten the lives of the residents of Sky Palace. Meliodas and his allies meet the Six Knights of Black in battle.',
        vote_average: 7,
        poster_path:
          'http://image.tmdb.org/t/p/original/fK6zf4eN5xxkCO3MP9rDNWOq6Ys.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 493922,
        title: 'Hereditary',
        release_date: '2018-06-04',
        overview:
          "When Ellen, the matriarch of the Graham family, passes away, her daughter's family begins to unravel cryptic and increasingly terrifying secrets about their ancestry. The more they discover, the more they find themselves trying to outrun the sinister fate they seem to have inherited.",
        vote_average: 7,
        poster_path:
          'http://image.tmdb.org/t/p/original/4GFPuL14eXi66V96xBWY73Y9PfR.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 363088,
        title: 'Ant-Man and the Wasp',
        release_date: '2018-07-04',
        overview:
          "As Scott Lang awaits expiration of his term of house detention, Hope van Dyne and Dr. Hank Pym involve him in a scheme to rescue Mrs. van Dyne from the micro-universe into which she has fallen, while two groups of schemers converge on them with intentions of stealing Dr. Pym's inventions.",
        vote_average: 7,
        poster_path:
          'http://image.tmdb.org/t/p/original/rv1AWImgx386ULjcf62VYaW8zSt.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 399360,
        title: 'Alpha',
        release_date: '2018-08-17',
        overview:
          'After a hunting expedition goes awry, a young caveman struggles against the elements to find his way home.',
        vote_average: 5.7,
        poster_path:
          'http://image.tmdb.org/t/p/original/afdZAIcAQscziqVtsEoh2PwsYTW.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 455980,
        title: 'Tag',
        release_date: '2018-05-30',
        overview:
          'For one month every year, five highly competitive friends hit the ground running in a no-holds-barred game of tag they’ve been playing since the first grade. This year, the game coincides with the wedding of their only undefeated player, which should finally make him an easy target. But he knows they’re coming...and he’s ready.',
        vote_average: 6.9,
        poster_path:
          'http://image.tmdb.org/t/p/original/eXXpuW2xaq5Aen9N5prFlARVIvr.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 500664,
        title: 'Upgrade',
        release_date: '2018-06-01',
        overview:
          'A brutal mugging leaves Grey Trace paralyzed in the hospital and his beloved wife dead. A billionaire inventor soon offers Trace a cure — an artificial intelligence implant called STEM that will enhance his body. Now able to walk, Grey finds that he also has superhuman strength and agility — skills he uses to seek revenge against the thugs who destroyed his life.',
        vote_average: 7.4,
        poster_path:
          'http://image.tmdb.org/t/p/original/woLGROUdPWdBnLVFKYdGD3kFkPY.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 429300,
        title: 'Adrift',
        release_date: '2018-05-31',
        overview:
          "A true story of survival, as a young couple's chance encounter leads them first to love, and then on the adventure of a lifetime as they face one of the most catastrophic hurricanes in recorded history.",
        vote_average: 6.4,
        poster_path:
          'http://image.tmdb.org/t/p/original/5gLDeADaETvwQlQow5szlyuhLbj.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 260513,
        title: 'Incredibles 2',
        release_date: '2018-06-14',
        overview:
          'Elastigirl springs into action to save the day, while Mr. Incredible faces his greatest challenge yet – taking care of the problems of his three children.',
        vote_average: 7.7,
        poster_path:
          'http://image.tmdb.org/t/p/original/x1txcDXkcM65gl7w20PwYSxAYah.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 447200,
        title: 'Skyscraper',
        release_date: '2018-07-11',
        overview:
          "Framed and on the run, a former FBI agent must save his family from a blazing fire in the world's tallest building.",
        vote_average: 6,
        poster_path:
          'http://image.tmdb.org/t/p/original/5LYSsOPzuP13201qSzMjNxi8FxN.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 345887,
        title: 'The Equalizer 2',
        release_date: '2018-07-19',
        overview:
          "Robert McCall returns to deliver his special brand of vigilante justice -- but how far will he go when it's someone he loves?",
        vote_average: 6.3,
        poster_path:
          'http://image.tmdb.org/t/p/original/cQvc9N6JiMVKqol3wcYrGshsIdZ.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 476292,
        title: 'Maquia: When the Promised Flower Blooms',
        release_date: '2018-02-24',
        overview:
          "A story of encounters and partings interwoven between people; this is a human drama with feelings that touch one's heart gradually, which everyone has experienced at least once.",
        vote_average: 5.1,
        poster_path:
          'http://image.tmdb.org/t/p/original/qcnOKCPleLOTWPPgYI0YT1MOQwR.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 442249,
        title: 'The First Purge',
        release_date: '2018-07-04',
        overview:
          'To push the crime rate below one percent for the rest of the year, the New Founding Fathers of America test a sociological theory that vents aggression for one night in one isolated community. But when the violence of oppressors meets the rage of the others, the contagion will explode from the trial-city borders and spread across the nation.',
        vote_average: 6,
        poster_path:
          'http://image.tmdb.org/t/p/original/2slvblTroiT1lY9bYLK7Amigo1k.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 487558,
        title: 'BlacKkKlansman',
        release_date: '2018-08-09',
        overview:
          'Ron Stallworth, an African-American police officer from Colorado, successfully manages to infiltrate the local Ku Klux Klan and become the head of the local chapter.',
        vote_average: 7.8,
        poster_path:
          'http://image.tmdb.org/t/p/original/55W6mUVv4CXMMQHHhV2zXtLSpXQ.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 487258,
        title: 'Marainthirunthu Paarkum Marmam Enna',
        release_date: '2018-08-17',
        overview:
          "A daring young man joins a gang that is behind chain-snatching crimes in the city. What's his game plan?",
        vote_average: 5.2,
        poster_path:
          'http://image.tmdb.org/t/p/original/2zfyj4gY7RHkf89ssSTly2UppMC.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 455207,
        title: 'Crazy Rich Asians',
        release_date: '2018-08-15',
        overview:
          "An American-born Chinese economics professor accompanies her boyfriend to Singapore for his best friend's wedding, only to get thrust into the lives of Asia's rich and famous.",
        vote_average: 7,
        poster_path:
          'http://image.tmdb.org/t/p/original/gnTqi4nhIi1eesT5uYMmhEPGNih.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 400155,
        title: 'Hotel Transylvania 3: Summer Vacation',
        release_date: '2018-06-30',
        overview:
          "Dracula, Mavis, Johnny and the rest of the Drac Pack take a vacation on a luxury Monster Cruise Ship, where Dracula falls in love with the ship's captain, Ericka, who's secretly a descendant of Abraham Van Helsing, the notorious monster slayer.",
        vote_average: 7,
        poster_path:
          'http://image.tmdb.org/t/p/original/gjAFM4xhA5vyLxxKMz38ujlUfDL.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      },
      {
        movie_id: 420814,
        title: 'Christopher Robin',
        release_date: '2018-08-02',
        overview:
          'Working-class family man Christopher Robin encounters his childhood friend Winnie-the-Pooh, who helps him to rediscover the joys of life.',
        vote_average: 7.3,
        poster_path:
          'http://image.tmdb.org/t/p/original/xR5w0he6czZkcAz459a4iPBqXGe.jpg',
        trailer: 'https://www.youtube.com/embed/wb49-oV0F78',
        favorite: false
      }
    ],
    favorites: [402900, 455980, 345887, 345940]
  },
  currentUser: { id: 2, name: 'asdf', password: 'asdf', email: 'asdf' }
};
