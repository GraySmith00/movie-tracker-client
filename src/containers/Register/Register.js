import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { registerUser } from '../../helpers/apiCalls';
import { setCurrentUser } from '../../actions/userActions';
import { setRegisterErrorState } from '../../actions/errorActions';
import './Register.css';

export class Register extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      password: ''
    };
  }

  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  handleSubmit = async e => {
    e.preventDefault();
    const { name, email, password } = this.state;
    const { setRegisterErrorState, history } = this.props;

    if (!name || !email || !password) {
      setRegisterErrorState('You are missing one or more required fields');
      return;
    }

    setRegisterErrorState('');
    try {
      const addedUser = await registerUser(this.state);

      if (!addedUser) {
        setRegisterErrorState('a user with this email address already exists');
        return;
      }

      this.setNewUserState(addedUser);

      if (addedUser) {
        history.push('/');
      }
    } catch (error) {
      setRegisterErrorState(error.message);
    }
  };

  setNewUserState = addedUser => {
    const { setCurrentUser } = this.props;
    setCurrentUser(addedUser);
    this.setState({
      name: '',
      email: '',
      password: ''
    });
  };

  render() {
    const { name, email, password } = this.state;

    return (
      <section className="register-user">
        <form onSubmit={this.handleSubmit} className="register-form">
          <h3 className="register-title">Sign up</h3>
          <input
            onChange={this.handleChange}
            value={name}
            name="name"
            type="text"
            placeholder="First Name"
            className="register-name"
          />
          <input
            onChange={this.handleChange}
            value={email}
            name="email"
            type="text"
            placeholder="Email"
            className="register-email"
          />
          <input
            onChange={this.handleChange}
            value={password}
            name="password"
            type="text"
            placeholder="Password"
            className="register-password"
          />
          <button className="register-btn">Submit</button>
        </form>
        <p className="error-message">{this.props.error}</p>
      </section>
    );
  }
}

Register.propTypes = {
  setCurrentUser: PropTypes.func.isRequired,
  error: PropTypes.string.isRequired
};

export const mapDispatchToProps = dispatch => ({
  setCurrentUser: user => dispatch(setCurrentUser(user)),
  setRegisterErrorState: message => dispatch(setRegisterErrorState(message))
});

const mapStateToProps = state => ({
  error: state.errors.registerError
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Register);
