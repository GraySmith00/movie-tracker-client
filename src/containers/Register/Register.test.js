import React from 'react';
import { shallow } from 'enzyme';
import { Register, mapDispatchToProps } from './Register';
import { mockStore } from '../../mockData/mockStore';
import configureMockStore from 'redux-mock-store';
import { setCurrentUser } from '../../actions/userActions';

describe('Register', () => {
  let wrapper;
  let store;
  let mockSetCurrentUser;
  let mockHistory;
  let mockSetRegisterErrorState;
  let mockRegisterError;
  const Store = configureMockStore();

  beforeEach(() => {
    mockHistory = jest.fn();
    mockSetCurrentUser = jest.fn();
    mockSetRegisterErrorState = jest.fn();
    mockRegisterError = '';

    store = Store(mockStore);
    wrapper = shallow(
      <Register
        store={store}
        setCurrentUser={mockSetCurrentUser}
        history={mockHistory}
        setRegisterErrorState={mockSetRegisterErrorState}
        error={mockRegisterError}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should setState on handleChange', () => {
    const mockNameEvent = { target: { name: 'name', value: 'paul' } };
    const mockEmailEvent = { target: { name: 'email', value: 'paul@paul' } };
    const mockPasswordEvent = {
      target: { name: 'password', value: 'paulword' }
    };

    wrapper.instance().handleChange(mockNameEvent);
    expect(wrapper.state('name')).toEqual(mockNameEvent.target.value);

    wrapper.instance().handleChange(mockEmailEvent);
    expect(wrapper.state('email')).toEqual(mockEmailEvent.target.value);

    wrapper.instance().handleChange(mockPasswordEvent);
    expect(wrapper.state('password')).toEqual(mockPasswordEvent.target.value);
  });

  it('should call setRegisterErrorState if name email or password are empty strings', () => {
    let mockEvent = { preventDefault: () => jest.fn() };

    wrapper.setState({ name: '', email: '', password: '' });
    wrapper.instance().handleSubmit(mockEvent);
    expect(mockSetRegisterErrorState).toHaveBeenCalledWith(
      'You are missing one or more required fields'
    );
  });

  it('should call setRegisterErrorState to set errors as an empty stringwith', () => {
    let mockEvent = { preventDefault: () => jest.fn() };
    wrapper.setState({ name: 'asdf', email: 'qwer', password: 'asdf' });

    wrapper.instance().handleSubmit(mockEvent);
    expect(mockSetRegisterErrorState).toHaveBeenCalledWith('');
  });

  it('should call setRegisterErrorState if email already exists', async () => {
    const mockResult = { error: 'Key (email)=(a) already exists.' };
    let mockEvent = { preventDefault: () => jest.fn() };

    wrapper.setState({ name: 'asdf', email: 'qwer', password: 'asdf' });
    window.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(mockResult)
      })
    );
    await wrapper.instance().handleSubmit(mockEvent);
    expect(mockSetRegisterErrorState).toHaveBeenCalledWith(
      'a user with this email address already exists'
    );
  });

  it('should call setCurrentUser and add user if user does not exist', () => {
    const mockEvent = { preventDefault: () => jest.fn() };
    const mockUser = { name: 'asdf', email: 'qwer', password: 'asdf' };

    const expected = [
      'http://localhost:3000/api/users/new/',
      {
        method: 'POST',
        body: JSON.stringify(mockUser),
        headers: {
          'Content-Type': 'application/json'
        }
      }
    ];

    wrapper.instance().handleSubmit(mockEvent);

    expect(window.fetch).toHaveBeenCalledWith(...expected);
  });

  it('should reset the state when a new user is registered', () => {
    const mockEvent

    const addedUser = {
      name: 'Paul',
      email: 'Paul@paul.com',
      password: 'password'
    };
    const expected = {
      name: '',
      email: '',
      password: ''
    };
    wrapper.setState({
      name: 'Paul',
      email: 'Paul@paul.com',
      password: 'password'
    });

    wrapper.instance().handleSubmit();

    expect(wrapper.state()).toEqual(expected);
  });

  it('should setState to empty strings on handleSubmit if register user', async () => {
    const mockEvent = { preventDefault: () => jest.fn() };
    const mockResult = {
      status: 'success',
      message: 'New user created',
      id: 49
    };
    window.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(mockResult)
      })
    );

    const expected = { email: '', name: '', password: '' };
    await wrapper.instance().handleSubmit(mockEvent);
    expect(wrapper.state()).toEqual(expected);
  });
  it('should route back to home page on user register', async () => {
    const mockEvent = { preventDefault: () => jest.fn() };
    const registerResult = {
      status: 'success',
      message: 'New user created',
      id: 49
    };

    window.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(registerResult)
      })
    );
    const x = await wrapper.instance().handleSubmit(mockEvent);
  });

  it('should have the store', () => {
    store.dispatch(setCurrentUser(mockStore.currentUser));
    const actions = store.getActions();
    const expectedPayload = [
      {
        type: 'SET_CURRENT_USER',
        user: { id: 2, name: 'asdf', password: 'asdf', email: 'asdf' }
      }
    ];
    expect(actions).toEqual(expectedPayload);
  });

  it('should call setRegisterErrorState if request is rejected', async () => {
    const mockEvent = { preventDefault: () => jest.fn() };

    window.fetch = jest
      .fn()
      .mockImplementation(() => Promise.reject(new Error('failed')));
    let expected = 'You are missing one or more required fields';
    await wrapper.instance().handleSubmit(mockEvent);

    await expect(mockSetRegisterErrorState).toHaveBeenCalledWith(expected);
    await expect(mockSetRegisterErrorState).toHaveBeenCalled();
  });

  it('should map to store on mapDispatchToProps with setCurrentUser', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.setCurrentUser();
    expect(mockDispatch).toHaveBeenCalled();
  });

  it('should map to store on mapDispatchToProps with setCurrentUser', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.setRegisterErrorState();
    expect(mockDispatch).toHaveBeenCalled();
  });
});
