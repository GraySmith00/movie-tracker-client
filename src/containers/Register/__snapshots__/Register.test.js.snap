// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register
    error=""
    history={[Function]}
    setCurrentUser={[Function]}
    setRegisterErrorState={[Function]}
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          className="register-form"
          onSubmit={[Function]}
>
          <h3
                    className="register-title"
          >
                    Sign up
          </h3>
          <input
                    className="register-name"
                    name="name"
                    onChange={[Function]}
                    placeholder="First Name"
                    type="text"
                    value=""
          />
          <input
                    className="register-email"
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    type="text"
                    value=""
          />
          <input
                    className="register-password"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="text"
                    value=""
          />
          <button
                    className="register-btn"
          >
                    Submit
          </button>
</form>,
        <p
          className="error-message"
/>,
      ],
      "className": "register-user",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="register-title"
>
              Sign up
</h3>,
            <input
              className="register-name"
              name="name"
              onChange={[Function]}
              placeholder="First Name"
              type="text"
              value=""
/>,
            <input
              className="register-email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
              value=""
/>,
            <input
              className="register-password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="text"
              value=""
/>,
            <button
              className="register-btn"
>
              Submit
</button>,
          ],
          "className": "register-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign up",
              "className": "register-title",
            },
            "ref": null,
            "rendered": "Sign up",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "register-name",
              "name": "name",
              "onChange": [Function],
              "placeholder": "First Name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "register-email",
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "register-password",
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "register-btn",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "error-message",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            className="register-form"
            onSubmit={[Function]}
>
            <h3
                        className="register-title"
            >
                        Sign up
            </h3>
            <input
                        className="register-name"
                        name="name"
                        onChange={[Function]}
                        placeholder="First Name"
                        type="text"
                        value=""
            />
            <input
                        className="register-email"
                        name="email"
                        onChange={[Function]}
                        placeholder="Email"
                        type="text"
                        value=""
            />
            <input
                        className="register-password"
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="text"
                        value=""
            />
            <button
                        className="register-btn"
            >
                        Submit
            </button>
</form>,
          <p
            className="error-message"
/>,
        ],
        "className": "register-user",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="register-title"
>
                Sign up
</h3>,
              <input
                className="register-name"
                name="name"
                onChange={[Function]}
                placeholder="First Name"
                type="text"
                value=""
/>,
              <input
                className="register-email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="text"
                value=""
/>,
              <input
                className="register-password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="text"
                value=""
/>,
              <button
                className="register-btn"
>
                Submit
</button>,
            ],
            "className": "register-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign up",
                "className": "register-title",
              },
              "ref": null,
              "rendered": "Sign up",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "register-name",
                "name": "name",
                "onChange": [Function],
                "placeholder": "First Name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "register-email",
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "register-password",
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "register-btn",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "error-message",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
